File: lesson_5/code/julila_set.py
import matplotlib.pyplot as plt
import numpy as np


points = 1000
iterations = 10
x = np.linspace(-2, 2, points)
y = np.linspace(-2, 2, points)
real, imag = np.meshgrid(x, y)
c = real + 1j * imag
z = c.copy()
escape_time = np.zeros_like(real)

plt.figure(figsize=(10, 10))
for _ in range(iterations):
    bounded = np.abs(z) <= 2
    z[bounded] = z[bounded] ** 2
    escape_time[bounded] += 1
    plt.clf()
    plt.imshow(escape_time, extent=[x.min(), x.max(), y.min(), y.max()], origin="lower", cmap="jet")
    plt.pause(0.5)

plt.show()

File: lesson_5/code/julila_set.m
   
    % Parameters
    points = 1000;
    i = 10; 

    % figure settings
    figure('Position', [100, 100, 1000, 1000]); 
    colormap([flipud(jet); 0 0 0]);  
    axis equal;

    % grid of complex numbers
    x = linspace(-2, 2, points);
    y = linspace(-2, 2, points);
    [real, complex] = meshgrid(x, y);
    C = real + complex * 1i;
    
    Z = C ;
    escape_time = zeros(size(C));
      
 
    % Mandelbrot calculation
    for n = 1:i
         % checks which points remain bounded 
        bounded = abs(Z) <= 2;
       
        % iterates bounded points
        Z(bounded) = Z(bounded).^2;

        % add time to bounded points
        escape_time(bounded) = escape_time(bounded) + 1;
        figure(1)
        imagesc(x, y, escape_time);

        pause(0.5)

    end
    